youtube_dl/extractor/mixcloud.py

```diff
@@ -122,3 +120,31 @@ def _real_extract(self, url):
             'view_count': view_count,
             'like_count': like_count,
         }
+
+class ServerAttemptSequence:
```

Comment:
The idea is cool, but it's too complicated. If I have read it correctly, you can use something simpler:
```python
def server_sequence(start, max_value):
     low = high = start
     yield start
     while low > 1 and high < max_value:
         low -= 1
         high += 1
         yield high
         yield low
```
Or even shorter:
```python
def server_sequence(start, max_value):
    yield start
    for low, high in zip(range(start - 1, 0, -1), range(start + 1, max_value)):
        yield high
        yield low
```
The last one may be a bit harder to understand, but I like it.